eth-providers/src/base-provider.ts

```diff
@@ -1772,6 +1775,28 @@ export abstract class BaseProvider extends AbstractProvider {
     return filteredLogs.map((log) => this.formatter.filterLog(log));
   };
 
+  _getPhantomLogsAtBlock = async (blockTag?: BlockTag | Promise<BlockTag>): Promise<any[]> => {
+    const blockHash = await this._getBlockHash(blockTag);
+    const blockNumber = (await this._getBlockHeader(blockHash)).number.toNumber();
+    const allEvents = await this.queryStorage<Vec<FrameSystemEventRecord>>('system.events', [], blockHash);
+
+    const evmLogs = allEvents
+      .filter(isPhantomEvmEvent)
+      .map(getPartialTransactionReceipt)
+      .map((receipt, i) =>
+        receipt.logs.map((log) => ({
+          ...log,
+          transactionIndex: 0,
+          transactionHash: keccak256(blockHash + i + i),
```

Comment:
why `+ i` twice? also this the hash of the string right? I think we want to hash the raw data rather hex string